### PR TITLE
Generate using statements inside namespaces

### PR DESCRIPTION
**Pull request by danlyons-softek**
This changes the stub generation's placement of using statements from the top of the file to inside the namespaces of each stub.  It is a quick-and-dirty means to address name collisions seen in Issue #784 

**Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: Refit.Tests/InterfaceStubGenerator.cs [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-using System.Text;
 using System.Threading.Tasks;
+
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Xunit;
-using Nustache;
-using Nustache.Core;
-using Refit; // InterfaceStubGenerator looks for this
```
Comment (onovotny) [inline Refit.Tests/InterfaceStubGenerator.cs]: Are the items in this file going to be skipped without this line? The generator looks for the `Using Refit` statement.

Comment (danlyons-softek) [reply Refit.Tests/InterfaceStubGenerator.cs]: Mea culpa - I have my editor set up to remove/sort using statements on save and I didn't notice the comment on that line.  I'll re-add it.

**Comment by onovotny** [comment]
Looks like an easy fix that'll capture many of the namespace clashes without requiring use of semantic analysis, thanks!

Please add a few tests that show this working (and to prevent regressions) and I can merge it.

**Comment by danlyons-softek** [comment]
I added a simple test for the `ClassTemplateInfo` objects generated for a couple interefaces referencing `SomeType` from the `CollisionA` and `CollisionB` namespaces.  Does that seem sufficient?

I could possibly add a test added to ensure the generated C# file compiles, but I'm not super familiar with Roslyn yet, and that may be too heavy-handed.

**Comment by onovotny** [comment]
That test is a part of it. I think a test that shows it working is useful too....so force the generated code to compensate for having two different `SomeType` instances in different namespaces, so we can prove the generated code works as intended.

I don't think you need to actually compile it as compiling the unit test project should trigger that code path.

**Comment by onovotny** [comment]
Thanks!

**Pull request merged**